src/fun/mod.rs

```diff
@@ -1026,7 +1032,8 @@ impl Definition {
   }
 
   pub fn new_gen(name: Name, rules: Vec<Rule>, builtin: bool) -> Self {
```

Comment:
I'm not exactly sure how this function is used, but generated functions should inherit the source of their generator.

So for example if we have
```
# File Nat/to_u24.bend
from Nat import {Succ, Zero}

def to_u24(x):
  match x:
    case Succ:
      return 1 + to_u24(x.pred)
    case Zero:
      return 0
```

This will generate function `Nat/to_u24__C0 = @x.pred (+ 1 (to_u24 x.pred))`.

The source of this generated function should be `Nat/to_u24.bend`, pointing at the `Succ` arm.

We don't need to carry the span to the generated function in this commit, that's a very large change for the future, but we could point it to the correct file at least.

I'm guessing that `new_gen` is used in this way.

Reply:
Fixed in the latest commit